### ISSUE TITLE
strEnum TypeError with non-string values doesn't raise an error or warning

### ISSUE THREAD

**Issue by Tomas542** [astral-sh/ruff#19386]
### Summary

I'm not sure if this is within the scope of `ruff`, so apologies if it isn't. The `strEnum` class is used to define enums with string attributes. So I think `ruff` should give a warning/error when we define a non-string attribute (which will cause a `TypeError` when the script runs). Both `ruff` and `pylance` (in the list with default rules) do nothing.

```python
from enum import StrEnum


class Foo(StrEnum):
    A = 1
    B = "aaa"


print(Foo("aaa"))
```
